src/System.Collections.Immutable/tests/ImmutableArrayTest.cs

```diff
@@ -251,7 +251,8 @@ public void Covariance()
             ImmutableArray<string> derivedImmutable = ImmutableArray.Create("a", "b", "c");
             ImmutableArray<object> baseImmutable = derivedImmutable.As<object>();
             Assert.False(baseImmutable.IsDefault);
-            Assert.Equal(derivedImmutable, baseImmutable);
+            // Must cast to object or the IEnumerable<object> overload of Equals would be used
+            Assert.Equal((object)derivedImmutable, baseImmutable);
```

Comment:
I think you're going to need to change these Equals to something like:
```C#
Assert.Equal((object)derivedImmutable, baseImmutable, EqualityComparer<object>.Default);
```
Otherwise, the default comparer that xunit uses will I believe compare the concrete types and fail the comparison as a result.